### ISSUE TITLE
[@types/mathjs] callback function return types in map function

### ISSUE THREAD

**Issue by appleparan** [DefinitelyTyped/DefinitelyTyped#30192]
If you know how to fix the issue, make a pull request instead.

- [x] I tried using the `@types/xxxx` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  (Please ask any appropriate questions there).
- [x] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @siavol @andnp @bradbesserman 

If you do not mention the authors the issue will be ignored.

Definition of `map` function shows return type of callback is `Matrix` or `MathArray`

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f01ba112a92d9b28d6a33ebadde37b29c74c3b18/types/mathjs/index.d.ts#L1348

```
callback: ((value: any, index: any, matrix: Matrix | MathArray) => Matrix | MathArray)
```

Compared to example of `mathjs` document, I'm not sure why return type is not a `Number`

```
math.map([1, 2, 3], function(value) {
  return value * value
})  // returns [1, 4, 9]
```
Is it right to add `Number` as a return type of callback?

**Comment by andnp** [comment]
Yup, I agree. This appears to be an oversight on our part. The correct definition should be:
```typescript
callback: (value: any, index: number, matrix: Matrix | MathArray) => number
```

Feel free to make a pull request to fix this if you'd like. If not, I can make one later (probably this weekend).
